components/badge/index.css

```diff
@@ -185,275 +84,270 @@
+	}
 
-.spectrum-Badge--blue {
-	background: var(--mod-badge-background-color-blue, var(--spectrum-badge-background-color-blue));
-}
+	&:where(.spectrum-Badge--cyan) {
+		background: var(--spectrum-cyan-background-color-default);
+	}
 
-.spectrum-Badge--indigo {
-	background: var(--mod-badge-background-color-indigo, var(--spectrum-badge-background-color-indigo));
-}
+	&:where(.spectrum-Badge--blue) {
+		background: var(--spectrum-blue-background-color-default);
+	}
 
-.spectrum-Badge--purple {
-	background: var(--mod-badge-background-color-purple, var(--spectrum-badge-background-color-purple));
-}
+	&:where(.spectrum-Badge--indigo) {
+		background: var(--spectrum-indigo-background-color-default);
+	}
 
-.spectrum-Badge--fuchsia {
-	background: var(--mod-badge-background-color-fuchsia, var(--spectrum-badge-background-color-fuchsia));
-}
+	&:where(.spectrum-Badge--purple) {
+		background: var(--spectrum-purple-background-color-default);
+	}
 
-.spectrum-Badge--magenta {
-	background: var(--mod-badge-background-color-magenta, var(--spectrum-badge-background-color-magenta));
-}
+	&:where(.spectrum-Badge--fuchsia) {
+		background: var(--spectrum-fuchsia-background-color-default);
+	}
 
-.spectrum-Badge--pink {
-	background: var(--mod-badge-background-color-pink, var(--spectrum-badge-background-color-pink));
-}
+	&:where(.spectrum-Badge--magenta) {
+		background: var(--spectrum-magenta-background-color-default);
+	}
 
-.spectrum-Badge--turquoise {
-	background: var(--mod-badge-background-color-turquoise, var(--spectrum-badge-background-color-turquoise));
-}
+	&:where(.spectrum-Badge--pink) {
+		background: var(--spectrum-pink-background-color-default);
+	}
 
-.spectrum-Badge--brown {
-	background: var(--mod-badge-background-color-brown, var(--spectrum-badge-background-color-brown));
-}
+	&:where(.spectrum-Badge--turquoise) {
+		background: var(--spectrum-turquoise-background-color-default);
+	}
 
-.spectrum-Badge--cinnamon {
-	background: var(--mod-badge-background-color-cinnamon, var(--spectrum-badge-background-color-cinnamon));
-}
+	&:where(.spectrum-Badge--brown) {
+		background: var(--spectrum-brown-background-color-default);
+	}
 
-.spectrum-Badge--silver {
-	background: var(--mod-badge-background-color-silver, var(--spectrum-badge-background-color-silver));
+	&:where(.spectrum-Badge--cinnamon) {
+		background: var(--spectrum-cinnamon-background-color-default);
+	}
+
+	&:where(.spectrum-Badge--silver) {
+		background: var(--spectrum-silver-background-color-default);
+	}
 }
 
 /* Style = subtle */
-.spectrum-Badge--style-subtle {
-	color: var(--mod-badge-subtle-label-icon-color, var(--spectrum-badge-subtle-label-icon-color));
+.spectrum-Badge--subtle {
+	color: var(--spectrum-gray-1000);
 
 	/* background color variants */
-	&.spectrum-Badge--neutral {
-		background: var(--mod-badge-subtle-background-color-default, var(--spectrum-badge-subtle-background-color-default));
+	&:where(.spectrum-Badge--neutral) {
```

Comment:
Tell me more about `:where()` in this context, it seems like the benefit over trying to match both classes is that it doesn't add specificity, right? I'm especially curious as we carry out similar CSS updates in other components. Will we want to consider this pattern in similar use cases in other components? Any pitfalls to be aware of? Any other things to note? (I believe it was `:where` in another component in that action menu PR I reviewed where we discovered that we actually did want the added specificity, for instance.)

Reply:
Yes, great questions. Badge is a good use-case for :where in that there are only a few variants active at any given time. I kept specificity on outline and subtle but where the colors showed up, I let the order they show up in the file be the deciding factor. For more complex components this gets ultra tricky because there are cases where we _want_ that specificity. This seemed like a great opportunity to reduce it though so I added it here.